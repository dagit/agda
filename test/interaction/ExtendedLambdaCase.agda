module ExtendedLambdaCase where

data Bool : Set where
  true false : Bool

data Void : Set where

foo : Bool -> Bool -> Bool -> Bool
foo = λ { x → λ { y z → {!!} } }

module parameterised {A : Set}(B : A -> Set) where

  data Bar : (Bool -> Bool) -> Set where
    baz : (t : Void) -> Bar λ { x → {!!} }

  -- with hidden argument
  data Bar' : (Bool -> Bool) -> Set where
    baz' : {t : Void} -> Bar' λ { x' → {!!} }


