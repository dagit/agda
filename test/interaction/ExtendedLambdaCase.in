top_command (cmd_load currentFile [])

-- Case splitting for function with implicit argument
top_command (showImplicitArgs True)
goal_command 3 cmd_make_case "z"
goal_command 3 cmd_make_case "w"
top_command (showImplicitArgs False)

-- Case splitting with lambda lifted definition, hidden arguments
goal_command 2 cmd_make_case "x'"
top_command (cmd_load currentFile [])
top_command (showImplicitArgs True)
goal_command 2 cmd_make_case "x'"


-- Case splitting with lambda lifted definition
goal_command 1 cmd_make_case "x"

-- Case splitting for nested extended lambdas
goal_command 0 cmd_make_case "y"

