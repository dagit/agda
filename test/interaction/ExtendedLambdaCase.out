(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-info-action "*Type-checking*" "Checking ExtendedLambdaCase (ExtendedLambdaCase.agda).\n" t)
(agda2-info-action "*Type-checking*" "Finished ExtendedLambdaCase.\n" t)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : Bool\n?1 : Bool\n?2 : Bool\n?3 : Bool\n" nil)
((last . 1) . (agda2-goals-action '(0 1 2 3)))
(agda2-status-action "ShowImplicit")
((last . 1) . (agda2-goals-action '(0 1 2 3)))
((last . 2) . (agda2-make-case-action-extendlam '("true {w} → ?" "false {w} → ?")))
((last . 1) . (agda2-goals-action '(0 1 2 3)))
((last . 2) . (agda2-make-case-action-extendlam '("z {true} → ?" "z {false} → ?")))
((last . 1) . (agda2-goals-action '(0 1 2 3)))
(agda2-status-action "")
((last . 1) . (agda2-goals-action '(0 1 2 3)))
((last . 2) . (agda2-make-case-action-extendlam '("true → ?" "false → ?")))
((last . 1) . (agda2-goals-action '(0 1 2 3)))
(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-info-action "*Type-checking*" "Checking ExtendedLambdaCase (ExtendedLambdaCase.agda).\n" t)
(agda2-info-action "*Type-checking*" "Finished ExtendedLambdaCase.\n" t)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : Bool\n?1 : Bool\n?2 : Bool\n?3 : Bool\n" nil)
((last . 1) . (agda2-goals-action '(0 1 2 3)))
(agda2-status-action "ShowImplicit")
((last . 1) . (agda2-goals-action '(0 1 2 3)))
((last . 2) . (agda2-make-case-action-extendlam '("true → ?" "false → ?")))
((last . 1) . (agda2-goals-action '(0 1 2 3)))
((last . 2) . (agda2-make-case-action-extendlam '("t true → ?" "t false → ?")))
((last . 1) . (agda2-goals-action '(0 1 2 3)))
((last . 2) . (agda2-make-case-action-extendlam '("x true z → ?" "x false z → ?")))
((last . 1) . (agda2-goals-action '(0 1 2 3)))
